File: src/prod/src/ServiceModel/EntryPointStatus.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

using namespace std;
using namespace Common;
using namespace ServiceModel;

std::wstring ServiceModel::EntryPointStatus::ToString(Enum const & val)
{
    wstring entryPointStatus;
    StringWriter(entryPointStatus).Write(val);
    return entryPointStatus;
}   


void ServiceModel::EntryPointStatus::WriteToTextWriter(TextWriter & w, Enum const & val)
{
    switch (val)
    {
        case Invalid: 
            w << L"Invalid";
            return;
        case Pending: 
            w << L"Pending";
            return;
        case Starting: 
            w << L"Starting";
            return;
        case Started: 
            w << L"Started";        
            return;
        case Stopping: 
            w << L"Stopping";
            return;
        case Stopped: 
            w << L"Stopped";
            return;
        default: 
            Common::Assert::CodingError("Unknown EntryPoint Status");
    }
}

FABRIC_ENTRY_POINT_STATUS EntryPointStatus::ToPublicApi(EntryPointStatus::Enum const & val)
{
    switch(val)
    {
        case Invalid: 
            return FABRIC_ENTRY_POINT_STATUS_INVALID;
        case Pending: 
            return FABRIC_ENTRY_POINT_STATUS_PENDING;
        case Starting: 
            return FABRIC_ENTRY_POINT_STATUS_STARTING;
        case Started: 
            return FABRIC_ENTRY_POINT_STATUS_STARTED;
        case Stopping: 
            return FABRIC_ENTRY_POINT_STATUS_STOPPING;
        case Stopped: 
            return FABRIC_ENTRY_POINT_STATUS_STOPPED;
        default: 
            Common::Assert::CodingError("Unknown Deployment Status");    
    }
}

ErrorCode EntryPointStatus::FromPublicApi(FABRIC_ENTRY_POINT_STATUS const & publicVal, __out Enum & val)
{
    switch(publicVal)
    {
    case FABRIC_APPLICATION_STATUS_INVALID:
        val = EntryPointStatus::Invalid;
        break;
    case FABRIC_ENTRY_POINT_STATUS_PENDING:
        val = EntryPointStatus::Pending;
        break;
    case FABRIC_ENTRY_POINT_STATUS_STARTING:
        val = EntryPointStatus::Starting;
        break;
    case FABRIC_ENTRY_POINT_STATUS_STARTED:
        val = EntryPointStatus::Started;
        break;
    case FABRIC_ENTRY_POINT_STATUS_STOPPING:
        val = EntryPointStatus::Stopping;
        break;
    case FABRIC_ENTRY_POINT_STATUS_STOPPED:
        val = EntryPointStatus::Stopped;
        break;
    default:
        return ErrorCode::FromHResult(E_INVALIDARG);
    }

    return ErrorCodeValue::Success;
}

File: src/prod/src/managed/Api/src/System/Fabric/EntryPointStatus.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric
{
    using System.Fabric.Interop;

        /// <summary>
        /// <para>Specifies the status of the CodePackage EntryPoint deployed on a node.</para>
        /// </summary>
    public enum EntryPointStatus
    {
        /// <summary>
        /// <para>The status of the entry point is unknown or invalid.</para>
        /// </summary>
        Invalid = NativeTypes.FABRIC_ENTRY_POINT_STATUS.FABRIC_ENTRY_POINT_STATUS_INVALID,
        /// <summary>
        /// <para>The entry point is scheduled to be started.</para>
        /// </summary>
        Pending = NativeTypes.FABRIC_ENTRY_POINT_STATUS.FABRIC_ENTRY_POINT_STATUS_PENDING,
        /// <summary>
        /// <para>The entry point is being started.</para>
        /// </summary>
        Starting = NativeTypes.FABRIC_ENTRY_POINT_STATUS.FABRIC_ENTRY_POINT_STATUS_STARTING,
        /// <summary>
        /// <para>The entry point was started successfully and is running.</para>
        /// </summary>
        Started = NativeTypes.FABRIC_ENTRY_POINT_STATUS.FABRIC_ENTRY_POINT_STATUS_STARTED,
        /// <summary>
        /// <para>The entry point is being stopped.</para>
        /// </summary>
        Stopping = NativeTypes.FABRIC_ENTRY_POINT_STATUS.FABRIC_ENTRY_POINT_STATUS_STOPPING,
        /// <summary>
        /// <para>The entry point is not running.</para>
        /// </summary>
        Stopped = NativeTypes.FABRIC_ENTRY_POINT_STATUS.FABRIC_ENTRY_POINT_STATUS_STOPPED,
    }
}